sources/packages/backend/apps/api/src/route-controllers/student/student.students.controller.e2e-spec.ts

```diff
@@ -1,113 +1,63 @@
-import * as faker from "faker";
-import { DatabaseModule, SINValidation, Student, User } from "@sims/sims-db";
-import {
-  StudentService,
-  UserService,
-  SINValidationService,
-} from "../../services";
+import { StudentService } from "../../services";
 import { KeycloakConfig } from "../../auth/keycloakConfig";
-import { KeycloakService } from "../../services/auth/keycloak/keycloak.service";
 import { HttpStatus, INestApplication } from "@nestjs/common";
-import { Test, TestingModule } from "@nestjs/testing";
 import * as request from "supertest";
-import { AuthModule } from "../../auth/auth.module";
-import { AppStudentsModule } from "../../app.students.module";
-import { createMockedZeebeModule } from "../../testHelpers/mocked-providers/zeebe-client-mock";
-import { NotificationsModule } from "@sims/services/notifications";
-import { MockedQueueModule } from "../../testHelpers/mocked-providers/queue-module-mock";
 import {
   ATBCCreateClientResponse,
   ATBCService,
 } from "@sims/integrations/services";
+import {
+  FakeStudentUsersTypes,
+  getStudentByFakeStudentUserType,
+  getStudentToken,
+  createTestingAppModule,
+  BEARER_AUTH_TYPE,
+} from "../../testHelpers";
+import { DataSource } from "typeorm";
 
 describe("Test ATBC Controller", () => {
-  const clientId = "student";
-  let accesstoken: string;
+  let accessToken: string;
   let app: INestApplication;
   let studentService: StudentService;
   let atbcService: ATBCService;
-  let userService: UserService;
-  let sinValidationService: SINValidationService;
+  let appDataSource: DataSource;
 
   beforeAll(async () => {
     await KeycloakConfig.load();
-    const token = await KeycloakService.shared.getToken(
-      process.env.E2E_TEST_STUDENT_USERNAME,
-      process.env.E2E_TEST_STUDENT_PASSWORD,
-      clientId,
+    accessToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
     );
-    accesstoken = token.access_token;
-    const moduleFixture: TestingModule = await Test.createTestingModule({
-      imports: [
-        DatabaseModule,
-        AuthModule,
-        AppStudentsModule,
-        NotificationsModule,
-        createMockedZeebeModule(),
-        MockedQueueModule,
-      ],
-    }).compile();
-    userService = await moduleFixture.get(UserService);
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    const moduleFixture = module;
```

Comment:
What is the need for this re-assignment?

Reply:
yes,  you're right. Not need for that. Thanks!